README.md

```diff
@@ -266,9 +266,13 @@ wss.on('connection', function connection(ws, request, client) {
 });
 
 server.on('upgrade', function upgrade(request, socket, head) {
+  /**
+   * NOTE:
+   *  The `authenticate` function not implemented! Please replace that by your auth logic!
+   */
   authenticate(request, (err, client) => {
     if (err || !client) {
-      socket.destroy();
+      socket.end('HTTP/1.1 401 Unauthorized');
```

Comment:
The user might want to reply with something different so I don't think this is better than current example where the socket is just destroyed. Also it is better to destroy the socket to prevent it from being kept indefinitely open if the other peer does not close it.

Reply:
I didn't know it before, thank you (I thought that 'end' will also destroy the socket). The problem I want to solve is sending a http response back to the browser, when the WebSocket can't be established.

Is this snippet better ?

```javascript
socket.end('HTTP/1.1 401 Unauthorized');
socket.destroy();

// or
socket.end('HTTP/1.1 401 Unauthorized');
setTimeout(function () {
  socket && socket.destroy();
},  500);
```

Reply:
```js
socket.write('HTTP/1.1 401 Unauthorized\r\n\r\n');
socket.destroy();
```

Reply:
Thanks

Reply:
Can you please apply this? I think it is reasonable since the function name is `authenticate`.